Fixed math notation in ABCI++ app requirements (#8499)
tendermint/tendermint: spec/abci++/abci++_app_requirements_002_draft.md

extensions will be discarded.

* Requirement 9 [`VerifyVoteExtension`, determinism-1]: `VerifyVoteExtension` is a (deterministic) function of
  the current state, the vote extension received, and the prepared proposal that the extension refers to.
  In other words, for any correct process *p*, and any arbitrary vote extension *e*, and any arbitrary
  block *w*, if *p*'s (resp. *q*'s) Tendermint calls `RequestVerifyVoteExtension` on *e* and *w* at height *h*,
  then *p*'s Application's acceptance or rejection **exclusively** depends on *e*, *w* and *s<sub>p,h-1</sub>*.

* Requirement 10 [`VerifyVoteExtension`, determinism-2]: For any two correct processes *p* and *q*,
  and any arbitrary vote extension *e*, and any arbitrary block *w*,
  if *p*'s (resp. *q*'s) Tendermint calls `RequestVerifyVoteExtension` on *e* and *w* at height *h*,
  then *p*'s Application accepts *e* if and only if *q*'s Application accepts *e*.
  Note that this requirement follows from Requirement 9 and the Agreement property of consensus.

Requirements 9 and 10 ensure that the validation of vote extensions will be deterministic at all
correct processes.

> Requirements 9 and 10 protect against arbitrary vote extension data from Byzantine processes

Reviewer: ```suggestion
Requirements 9 and 10 protect against arbitrary vote extension data from Byzantine processes,
```

> similarly to Requirements 6 and 7 protecting against arbitrary proposed blocks.

Reviewer: ```suggestion
in a similar way as Requirements 6 and 7 protect against arbitrary proposed blocks.
```

Requirements 9 and 10 can be violated by a bug inducing non-determinism in
`VerifyVoteExtension`. In this case liveness can be compromised.
Extra care should be put in the implementation of `ExtendVote` and `VerifyVoteExtension` and,

> as a general rule, `VerifyVoteExtension` SHOULD always accept the vote extension.

Reviewer: ```suggestion
As a general rule, `VerifyVoteExtension` SHOULD always accept the vote extension.
```
